src/java/org/apache/cassandra/index/sai/disk/v1/InvertedIndexSearcher.java

```diff
@@ -248,7 +248,7 @@ private CloseableIterator<PrimaryKeyWithSortKey> bm25Internal(CloseableIterator<
         // cap frequencies to total rows so that the IDF term doesn't turn negative
         var cappedFrequencies = documentFrequencies.entrySet().stream()
                                                     .collect(Collectors.toMap(Map.Entry::getKey, e -> Math.min(e.getValue(), totalRows)));
-        var docStats = new BM25Utils.DocStats(cappedFrequencies, totalRows);
+        BM25Utils.DocStats docStats = new BM25Utils.DocStats(cappedFrequencies, totalRows, metadata.termsDistribution.numPoints);
```

Comment:
The `numPoints` appears to be a count of unique term/row pairs, which means documents with terms that have frequency > 1 will get under-counted in the doc length computation, which seems problematic. I think we'll need to store the total number of terms inserted into the index when we build the index via the memtable or via compaction, and then we can store the value in the index's metadata or in the `termsDistribution` object (I'm not sure what makes the most sense).

Most importantly, I think this will matter more on small test datasets where users might be checking the implementation.

Reply:
>The numPoints appears to be a count of unique term/row pairs, which means documents with terms that have frequency > 1 will get under-counted in the doc length computation, which seems problematic. I think we'll need to store the total number of terms inserted into the index when we build the index via the memtable or via compaction, and then we can store the value in the index's metadata or in the termsDistribution object (I'm not sure what makes the most sense).

Thank you for noticing that I used incorrect count, `numPoints`. I will add storing the total term count. (and I fixed `numRows`, which wasn't correct, #1695)